File: backend/app/schemas/assistant.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field

class ChatRequest(BaseModel):
    message: str
    video_id: Optional[int] = None
    event_id: Optional[int] = None

class ChatResponse(BaseModel):
    reply: str
    grounding_data: Optional[Dict[str, Any]] = None
    citations: List[str] = Field(default_factory=list)
    confidence: float = 0.95

class IncidentExplainRequest(BaseModel):
    event_id: int

class IncidentExplainResponse(BaseModel):
    event_id: int
    what_happened: str
    why_risky: str
    evidence: List[str]
    sop_rule: str
    sop_rule_name: str
    recommendation: str
    full_explanation: str

class ShiftSummaryRequest(BaseModel):
    video_id: Optional[int] = None

class ShiftSummaryResponse(BaseModel):
    shift_title: str
    total_events: int
    risk_events: int
    critical: int
    high: int
    medium: int
    low: int
    most_frequent_behaviour: str
    highest_risk_incident: str
    highest_risk_zone: str
    recurring_patterns: List[str]
    recommendations: List[str]
    full_summary_text: str

File: frontend/src/api/assistant.ts
import { useMutation } from '@tanstack/react-query';
import { apiRequest } from './client';

export interface ChatPayload {
  message: string;
  video_id?: number;
  event_id?: number;
}

export function useSendAssistantMessage() {
  return useMutation({
    mutationFn: async (payload: ChatPayload) => {
      return apiRequest<{
        reply: string;
        citations: string[];
        confidence: number;
      }>('/assistant/chat', {
        method: 'POST',
        body: JSON.stringify(payload)
      });
    }
  });
}

export function useExplainIncident() {
  return useMutation({
    mutationFn: async (eventId: number) => {
      return apiRequest<{
        event_id: number;
        what_happened: string;
        why_risky: string;
        evidence: string[];
        sop_rule: string;
        sop_rule_name: string;
        recommendation: string;
        full_explanation: string;
      }>('/assistant/explain', {
        method: 'POST',
        body: JSON.stringify({ event_id: eventId })
      });
    }
  });
}

export function useGenerateShiftSummary() {
  return useMutation({
    mutationFn: async (videoId?: number) => {
      return apiRequest<{
        shift_title: string;
        total_events: number;
        risk_events: number;
        critical: number;
        high: number;
        medium: number;
        low: number;
        most_frequent_behaviour: string;
        highest_risk_incident: string;
        highest_risk_zone: string;
        recurring_patterns: string[];
        recommendations: string[];
        full_summary_text: string;
      }>('/assistant/summary', {
        method: 'POST',
        body: JSON.stringify({ video_id: videoId })
      });
    }
  });
}
